src/API/Authentication.php

```diff
@@ -499,11 +505,46 @@ public function dbConnectionsChangePassword(
             method('post')->
             addPath('dbconnections', 'change_password')->
             withBody(Toolkit::merge([
-                'client_id'  => $this->getConfiguration()->getClientId(\Auth0\SDK\Exception\ConfigurationException::requiresClientId()),
+                'client_id'  => $this->getConfiguration()->getClientId(ConfigurationException::requiresClientId()),
                 'email'      => $email,
                 'connection' => $connection,
             ], $body))->
             withHeaders($headers)->
             call();
     }
+
+    /**
+     * Add client authentication to a request body.
+     *
+     * @param array<mixed> $requestBody Request body being sent to the endpoint.
+     *
+     * @return array<mixed> Request body with client authentication added.
+     *
+     * @throws TokenException If client assertion signing key or algorithm is invalid.
+     * @throws ConfigurationException If client ID or secret are invalid.
+     */
+    private function addClientAuthentication(array $requestBody): array
+    {
+        $clientAssertionSigningKey = $this->getConfiguration()->getClientAssertionSigningKey();
+        $clientAssertionSigningAlgorithm = $this->getConfiguration()->getClientAssertionSigningAlgorithm();
+
+        $requestBody['client_id'] = $this->getConfiguration()->getClientId(ConfigurationException::requiresClientId()) ?? '';
+
+        if (null !== $clientAssertionSigningKey) {
+            $requestBody['client_assertion_type'] = self::CONST_CLIENT_ASSERTION_TYPE;
+            $requestBody['client_assertion'] = ClientAssertionGenerator::create(
+                domain: $this->getConfiguration()->formatDomain(true) . '/',
+                clientId: $this->getConfiguration()->getClientId(ConfigurationException::requiresClientId()) ?? '',
+                signingKey: $clientAssertionSigningKey,
+                signingAlgorithm: $clientAssertionSigningAlgorithm
+            );
+
+            return $requestBody;
+        }
+
+        $clientSecret = $this->getConfiguration()->getClientSecret(ConfigurationException::requiresClientSecret()) ?? '';
```

Comment:
You should still be able to do things like code exchange with PKCE without any client authentication, it looks like this will fail with client secret required now?

Reply:
Resolved per our Slack chat 👍